klaviyo/__init__.py

```diff
@@ -76,11 +91,188 @@ def identify(self, email=None, id=None, properties=None, is_test=False):
+            'sort': sort,
+            'since': since
+        }
+        params = self._filter_params(params)
+        
+        if metric_id:
+            url = '{}/{}/{}'.format(METRIC, metric_id, TIMELINE)
+        else:
+            url = '{}/{}'.format(METRICS, TIMELINE)
+
+        timeline = self._request(url, params)
+        
+        return timeline
     
+    def metric_export(
+        self, 
+        metric_id, 
+        start_date=None, 
+        end_date=None, 
+        unit=None, 
+        measurement=None, 
+        where=None, 
+        by=None, 
+        count=None
+        ):
+        params = {
+            'metric_id': metric_id,
+            'start_date': start_date,
+            'end_date': end_date, 
+            'unit': unit,
+            'measurement': measurement,
+            'where': where,
+            'by': by,
+            'count': count
+        }
+        params = self._filter_params(params)
+        
+        url = '{}/{}/{}'.format(METRIC, metric_id, 'export')
+        
+        metric_export = self._request(url, params)
+        
+        return metric_export
+
+    def lists(self, list_name=None, method='GET'):
+        """
+        args:
+            method: str() optional type of request
+            list_name: str() optional name of list to be created
+        """
+        api_version = 'v2'
+        if method.upper() == 'GET':
+            lists = self._request('lists', {}, api_version=api_version)
+            return lists
+
+        elif method.upper() == 'POST':
+            params = {
+                'list_name': list_name
+            }
+            created_list = self._request('lists', params, method=method, api_version=api_version)
+            return created_list
+        
+    def list(self, list_id, list_name=None, method="GET",):
+        """
+        args:
+            list_id: str() the list id
+            list_name: str() name of the list - PUT to change it
+        """
+        if method == 'POST':
+            raise KlaviyoException('The list endpoint only accepts GET, PUT, and DELETE methods')
+        api_version = 'v2'
+        params = {
+            "list_name": list_name
+        }
+        params = self._filter_params(params)
+        list_details = self._request('list/{}'.format(list_id), params, method=method, api_version=api_version)
+
+        return list_details
+        
+    def list_subscription(self, list_id, subscription_type, data, method="GET"):
+        """
+        args:
+            list_id: str() the list id
+            subscription_type: str() subscribe or members depending on the action
+            data: for POST -> data must be a list of objects, for GET data must be a list of emails
+            
+        """
+        api_version = 'v2'
+        subscription_type = subscription_type.lower()
+        if subscription_type not in SUBSCRIPTION_MEMBERS:
+            raise KlaviyoException('The subscription type must be subscribe or members')
```

Comment:
I suppose having SUBSCRIPTION_MEMBERS as a global variable makes it easy for you to append to it if the API was extended.  Its gonna bite you with this Exception message since it can easily be forgotten to update it as well.  Hence I think its better without the global variables

Reply:
Thoughts on defaulting to `'subscribe'`? then you can change to the `members` if you want?

Reply:
not a bad idea.  Since we would be able to call 'members' if we do it explicitly